prime-router/docs/design/design/mapping-schemas.md

```diff
@@ -0,0 +1,77 @@
+# Mapping schema file structure
+
+Translation between HL7v2 and FHIR requires schema files that define how data is translated from one format to another. As we have built out these mappings, shortcomings in the current layout of the files have become apparent. This proposed structure aims to be self-documenting and logical so that the mappings are more maintainable and reusable.
+
+## HL7v2 to FHIR mapping file organization
+
+### Legacy organization
+
+Currently the schemas are organized by the FHIR resource or datatype the HL7v2 data is mapped to, like so:
+
+- hl7/message/ORU_R01.yml
+- hl7/resource/MessageHeader.yml
+- hl7/datatype/Identifier.yml
+- hl7/datatype/Identifer_XON.yml
+
+### Updated organization
+
+A self-documenting directory structure would allow the developer to see the source HL7v2 segment or datatype and the target FHIR mapping from its name and location alone. One such proposed format could implement the following structure:
+
+- hl7/messages/_**[HL7v2 message type]**_/  ← message types, such as ADT_A01, ORU_R01, etc.
+- hl7/segments/_**[HL7v2 segment]**_/_**[FHIR path]**_
+- hl7/segments/_**[HL7v2 segment]**_/_**[HL7v2 datatype]**_/_**[FHIR path]**_ ← for segment specific datatype definitions
+- hl7/datatypes/_**[HL7v2 datatype]**_/_**[FHIR path]**_ ← for standard datatype mappings
+
+For message types, each message type would have a directory named after itself. Ideally, the only content within that directory should be a schema that references the other top level segments and datatypes. If there are segments or datatypes that need to be defined differently from the standard, then this directory would have its own segments and datatypes directories as needed to store schemas specific to this message type.
+
+- hl7/messages/ADT_A01/ADT_A01.yml
+- hl7/messages/ADT_A01/resources ← only if base level resource cannot be used
```

Comment:
Nit: `resources` doesn't seem to line up with the other names.  Should this maybe just be a `segments` and `datatypes` directories?